core/controllers/reader.py

```diff
@@ -79,26 +79,62 @@ def _does_exploration_exist(exploration_id, version, collection_id):
 class ExplorationEmbedPage(base.BaseHandler):
     """Page describing a single embedded exploration."""
 
+    URL_PATH_ARGS_SCHEMAS = {
+        'exploration_id': {
+            'schema': editor.SCHEMA_FOR_EXPLORATION_ID
+        }
+    }
+    HANDLER_ARGS_SCHEMAS = {
+        'GET': {
+            'v': {
+                'schema': {
+                    'type': 'int',
+                    'validators': [{
+                        'id': 'is_at_least',
+                        'min_value': 1
+                    }]
+                },
+                'default_value': None
+            },
+            'collection_id': {
+                'schema': {
+                    'type': 'basestring',
+                    'validators': [{
+                        'id': 'is_regex_matched',
+                        'regex_pattern': constants.ENTITY_ID_REGEX
+                    }]
+                },
+                'default_value': None
+            },
+            'iframed': {
+                'schema': {
+                    'type': 'bool'
+                },
+                'default_value': False
+            },
+        }
+    }
+
     @acl_decorators.can_play_exploration
     def get(self, exploration_id):
         """Handles GET requests.
 
         Args:
             exploration_id: str. The ID of the exploration.
         """
-        version_str = self.request.get('v')
+        version_str = self.normalized_request.get('v')
         version = int(version_str) if version_str else None
 
         # Note: this is an optional argument and will be None when the
         # exploration is being played outside the context of a collection.
-        collection_id = self.request.get('collection_id')
+        collection_id = self.normalized_request.get('collection_id', None)
```

Comment:
Why is `None` added?

Reply:
If collection_id not present in the parameters, then the validator would not populate it because the default value is None. In order to be able to access it, None is added so that None will be taken

Reply:
Okay